SC_redesign/Crop_and_Soil/soil/carbon_cycling/pool_gas_partition.py

```diff
@@ -0,0 +1,626 @@
+import math
+from typing import Optional, List
+from SC_redesign.Crop_and_Soil.soil.soil_data import SoilData
+
+"""
+This module contains all necessary methods for updating all kinds of carbon in the field, based on the
+pseudocode soil documentation.
+"""
+
+
+class PoolGasPartition:
+    def __init__(self, soil_data: Optional[SoilData] = None):
```

Comment:
Add class docstrings here and elsewhere:

```suggestion
class PoolGasPartition:
    """[Add description of what the class is for]
    
    Attributes
    ----------
    data : SoilData, optional
        a reference to the soil data object, which will be used and updated. If not given, a default SoilData
        object is created.
    """
    def __init__(self, soil_data: Optional[SoilData] = None):
```